src/modules/common/components/__tests__/FailureSwitcher.spec.js

```diff
@@ -1,44 +1,49 @@
 // @flow
 
 import React from 'react'
-import { shallow } from 'enzyme'
+import { mount } from 'enzyme'
 
 import { FailureSwitcher } from '../FailureSwitcher'
 import ContentNotFoundError from '../../errors/ContentNotFoundError'
+import { Failure } from '../Failure'
 
 describe('FailureSwitcher', () => {
   const city = 'augsburg'
 
   const language = 'de'
 
-  const mockTranslate = key => key || 'null'
-
-  describe('render component not found failure', () => {
-    it('should render a category not found failure and match snapshot', () => {
-      const error = new ContentNotFoundError({ type: 'category', id: 'willkommen', language, city })
-      const renderContentNotFoundComponent = FailureSwitcher.renderContentNotFoundComponent(error)
-      expect(renderContentNotFoundComponent).toMatchSnapshot()
+  describe.each`
+    type                | id
+    ${'category'}       | ${'willkommen'}
+    ${'event'}          | ${'1234'}
+    ${'localNewsItem'}  | ${'/augsburg/en/news/local/1'}
+    ${'tunewsItem'}     | ${'/augsburg/en/news/tu-news/1'}
+    ${'extra'}          | ${'sprungbrett'}
+    ${'poi'}            | ${'1234'}
+    `('render $type component not found failure',
+  ({ type, id }) => {
+    it(`should render a ${type} not found failure and match snapshot`, () => {
+      const error = new ContentNotFoundError({ type, id, language, city })
+      const component = FailureSwitcher.renderContentNotFoundComponent(error)
+      expect(component).toMatchSnapshot()
```

Comment:
Maybe a `.dive()` would be cool here. I think then the Translate HOC would be gone in the tests.

Reply:
does not work, as dive is an enzyme functionality and the component is not rendered with shallow or mount. The current solution definitly looks cleaner, than a render tree (even with shallow)